### DISCUSSION TITLE
Cannot replace %20 in browser url

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#49570] [Help] [answered]
### Summary

When tab change I replace browser hash . and there I see %20 instead of space. I tried encodeURL and decode none of them didn't change . Also replace didn't work . Chatgpt said add delay and it also didn't work .
original tab = new 33 rr r
in browser = http://localhost:5173/#new%2033%20rr%20r

![Screenshot (146)](https://github.com/vercel/next.js/assets/71011043/ac22b7b7-175c-4e2e-ad0c-357c5ad42d08)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is part of the URL specification. That is, the browser is doing this because that's how the URL standard works.

URLs cannot contain spaces. Instead they are encoded into `%20`. 

You have `tab.replace` on line 29 of the screenshot, but I guess you want to replace `" "` with `"-"`? Either way, you can't have spaces there, too bad.

In other words, React is not changing the spaces by `%20` (I think you are assuming that) instead, when you do `location.replace` the browser encodes the space into `%20`:

```js
window.location.replace("#hello world")
```

Results in: https://example.com/#hello%20world

**Reply by thanhtutzaw** [reply]
so may be I should replace space with - before changing url . Thank you bro .

**Reply by thanhtutzaw** [reply]
Now it works . I do it with regex (answer from stackoverflow) and didn't  work . Now I use replaceAll .
![Screenshot (147)](https://github.com/vercel/next.js/assets/71011043/a8cbe58b-2487-4e27-a3d3-01af8a72abef)
